test/e2e/operator_test.go

```diff
@@ -233,163 +233,173 @@ func TestMain(m *testing.M) {
-		t.Fatalf("Unable to create a pod: %v", err)
-	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// runoncedurationoverrides.admission.runoncedurationoverride.openshift.io/enabled: "true"
+			ctx, cancelFnc := context.WithCancel(context.TODO())
+			defer cancelFnc()
 
-	if err := wait.PollImmediate(1*time.Second, time.Minute, func() (bool, error) {
-		klog.Infof("Listing pods...")
-		pod, err := clientSet.CoreV1().Pods(pod.Namespace).Get(ctx, pod.Name, metav1.GetOptions{})
-		if err != nil {
-			klog.Errorf("Unable to get pod: %v", err)
-			return false, nil
-		}
-		if pod.Spec.NodeName == "" {
-			klog.Infof("Pod not yet assigned to a node")
-			return false, nil
-		}
-		klog.Infof("Pod successfully assigned to a node: %v", pod.Spec.NodeName)
+			clientSet := getKubeClientOrDie()
 
-		if pod.Spec.ActiveDeadlineSeconds != nil && *pod.Spec.ActiveDeadlineSeconds == 3600 {
-			klog.Infof("pod.Spec.ActiveDeadlineSeconds is set to 3600, even though restartPolicy is Always")
-			return false, nil
-		}
+			testNamespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "e2e-" + strings.ToLower(t.Name()),
+					Labels: map[string]string{
+						"runoncedurationoverrides.admission.runoncedurationoverride.openshift.io/enabled": "true",
+					},
+				},
+			}
+			if _, err := clientSet.CoreV1().Namespaces().Create(ctx, testNamespace, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("Unable to create ns %v", testNamespace.Name)
+			}
+			defer clientSet.CoreV1().Namespaces().Delete(ctx, testNamespace.Name, metav1.DeleteOptions{})
+			test.pod.Namespace = testNamespace.Name
 
-		return true, nil
-	}); err != nil {
-		t.Fatalf("Unable to wait for a scheduled pod: %v", err)
-	}
+			if _, err := clientSet.CoreV1().Pods(test.pod.Namespace).Create(ctx, test.pod, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("Unable to create a pod: %v", err)
+			}
 
+			if err := wait.PollImmediate(1*time.Second, time.Minute, func() (bool, error) {
+				klog.Infof("Listing pods...")
+				pod, err := clientSet.CoreV1().Pods(test.pod.Namespace).Get(ctx, test.pod.Name, metav1.GetOptions{})
+				if err != nil {
+					klog.Errorf("Unable to get pod: %v", err)
+					return false, nil
+				}
+				if pod.Spec.NodeName == "" {
+					klog.Infof("Pod not yet assigned to a node")
+					return false, nil
+				}
+				klog.Infof("Pod successfully assigned to a node: %v", pod.Spec.NodeName)
+
+				if test.expectedToSet {
+					if pod.Spec.ActiveDeadlineSeconds == nil || *pod.Spec.ActiveDeadlineSeconds != test.activeDeadlineSeconds {
+						klog.Infof("pod.Spec.ActiveDeadlineSeconds is not set to 3600")
+						return false, nil
+					}
+				} else {
+					if pod.Spec.ActiveDeadlineSeconds != nil && *pod.Spec.ActiveDeadlineSeconds == test.activeDeadlineSeconds {
+						klog.Infof("pod.Spec.ActiveDeadlineSeconds is set to 3600, even though restartPolicy is Always")
```

Comment:
"pod.Spec.ActiveDeadlineSeconds is set to 3600, even though restartPolicy is Always" -> "pod.Spec.ActiveDeadlineSeconds is set to %v, even though restartPolicy is Always", test.activeDeadlineSeconds